File: flink_java/src/main/java/github/yahuili1128/streamOperators/UnionDemo.java
package github.yahuili1128.streamOperators;

import github.yahuili1128.pojo.MockUpModel;
import org.apache.flink.streaming.api.datastream.DataStream;
import org.apache.flink.streaming.api.datastream.SingleOutputStreamOperator;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;

import static github.yahuili1128.connector.SourceKafka010.getMockUpkafka010;

/**
 * @Author: LiYahui
 * @Date: Created in  2019/7/14 17:17
 * @Description: TODO flink union java
 * @Version: V1.0
 */
public class UnionDemo {
	public static void main(String[] args) throws Exception {
		final StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
		SingleOutputStreamOperator<MockUpModel> kafka010Data = getMockUpkafka010(env);
		DataStream<MockUpModel> unionDS = kafka010Data.union(kafka010Data,kafka010Data);
		unionDS.countWindowAll(5).max("age")
				.print().setParallelism(1);
		env.execute("kafka010 demo");
	}
}

File: flink_scala/src/main/scala/github/yahuili1128/streamOperators/UnionDemo.scala
package github.yahuili1128.streamOperators

import github.yahuili1128.connector.ScalaSourceKafka010
import org.apache.flink.streaming.api.scala._
import org.apache.flink.streaming.api.windowing.time.Time

/**
  * @Description : flink union scala
  * @Author : LiYahui
  * @Date : 2019-07-15 18:21
  * @Version : V1.0
  */
object UnionDemo {
	def main(args: Array[String]): Unit = {
		val env: StreamExecutionEnvironment = StreamExecutionEnvironment.getExecutionEnvironment
		val mockup: DataStream[ScalaSourceKafka010.MockUp] = ScalaSourceKafka010.getMockUpFromKafka(env)
		val unionDS: DataStream[ScalaSourceKafka010.MockUp] = mockup.union(mockup, mockup)
		unionDS.print().setParallelism(1)


		env.execute(s"${this.getClass.getSimpleName}")
	}

}
